tests/components/network_sediment_transporter/test_bed_initializer.py

```diff
@@ -0,0 +1,272 @@
+
+
+def test_call_depth_BPI(example_nmg2):
+    depth_at_link = np.full(example_nmg2.number_of_links, 10.0)  # m
+    initialize_parcels = BedParcelInitializerDepth(
+        example_nmg2,
+        flow_depth_at_link=depth_at_link,
+    )
+    _ = initialize_parcels()
+
+
+def test_call_userD50_BPI(example_nmg2):
+    d50_at_link = np.full(example_nmg2.number_of_links, 0.08)  # m
+    initialize_parcels = BedParcelInitializerUserD50(
+        example_nmg2,
+        user_d50=d50_at_link,
+    )
+    _ = initialize_parcels()
+
+
+# %% Tests of BPI base class helper functions
+
+
+def test_calc_total_parcel_volume():
+    vole = 12.5
+    vol = calc_total_parcel_volume(5, 5, 0.5)
+    np.testing.assert_allclose(vol, vole, rtol=1e-4)
+
+
+def test_det_approx_parcel_volume():
+    total_parcel_volume_at_link = 10
+    median_number_of_starting_parcels = 50
+    pvole = 0.2
+    pvol = _determine_approx_parcel_volume(
+        total_parcel_volume_at_link, median_number_of_starting_parcels
+    )
+    np.testing.assert_allclose(pvol, pvole, rtol=1e-4)
+
+
+# %% Test to make sure we're getting correct correct d50 correct_values
+
+
+def test_calc_D50_discharge():
+    """test calc D50 grain size give correct values"""
+    width = 20
+    slope = 0.01
+    discharge = 100
+    mannings_n = 0.05
+    gravity = 9.81
+    rho_water = 1000
+    rho_sediment = 3000
+    tau_c_50 = 0.05
+
+    D50e = 0.173286
+
+    D50 = calc_d50_discharge(
+        width,
+        slope,
+        discharge,
+        mannings_n,
+        gravity,
+        rho_water,
+        rho_sediment,
+        tau_c_50,
+    )
+
+    np.testing.assert_allclose(D50, D50e, rtol=1e-4)
+
+
+class Test_calc_d50_dArea_scaling(object):
+    def test_normal_d50_dArea_scaling(self):
+        """test normal values 1"""
+        D50e = 0.13182567
+        a = 0.1
+        n = 0.12
+        CA = 10  # km^2
+        D50 = calc_d50_dArea_scaling(drainage_area=CA, a=a, n=n)
+        np.testing.assert_allclose(D50, D50e, rtol=1e-4)
+
+    def test_normal_d50_dArea_scaling_2(self):
+        """test normal values 2"""
+        D50e = 0.1
+        a = 0.1
+        n = 0
+        CA = 10  # km^2
+        D50 = calc_d50_dArea_scaling(drainage_area=CA, a=a, n=n)
+        np.testing.assert_allclose(D50, D50e, rtol=1e-4)
+
+
+def test_calc_d50_depth():
+    """test normal values for calc_d50_depth"""
+    D50e = 0.1
+    D50 = calc_d50_depth(
+        slope=0.01,
+        flow_depth=1,
+        tau_c_multiplier=1,
+        rho_water=1000.0,
+        rho_sediment=3000.0,
+        tau_c_50=0.05,
+    )
+    np.testing.assert_allclose(D50, D50e, rtol=1e-4)
+
+
+# %% Test for errors raised
+def test_D50_not_specified(example_nmg2):
+    """test ValueError exception raised if D50 input is None"""
+    with pytest.raises(ValueError):
+        initialize_parcels = BedParcelInitializerUserD50(
+            example_nmg2,
+            user_d50=None,
+        )
+        _ = initialize_parcels()
```

Comment:
You don't need the `_ =` for any of these.

Reply:
removed `_ =` here and elsewhere